Admin generated url issue

MarekB01 opened the pull request:
fixes: https://github.com/BroadleafCommerce/QA/issues/5001

Rewrote convertToUrlFragment, so now if product name contains special symbols and "." between it, it'll be removed in generated url field

MarekB01 committed:
BroadleafCommerce/QA-5001_generated_url_issue

Rewrote convertToUrlFragment, so now if product name contains special symbols and "." between it, it'll be removed in generated url field

MarekB01 committed:
BroadleafCommerce/QA-5001_generated_url_issue

Rewrote convertToUrlFragment, so now if product name contains special symbols and "." between it, it'll be removed in generated url field. Updated code to not remove the DOT from the product name.

agustinlacombadev reviewed (commented):
I think that in the Old code the space was replaced by '-' , like this 
![image](https://github.com/BroadleafCommerce/BroadleafCommerce/assets/78038516/ae24b498-98d2-4a11-ad35-3c63ef7321e5)

but now the Space is removed. after '.'
![image](https://github.com/BroadleafCommerce/BroadleafCommerce/assets/78038516/1adb91bf-dc23-41b1-837e-1188ec6880ee)

MarekB01 committed:
BroadleafCommerce/QA-5001_generated_url_issue

Rewrote convertToUrlFragment, so now if product name contains special symbols and "." between it, it'll be removed in generated url field. Updated code to not remove the DOT from the product name. Refactored code, updated it due to requirements

MarekB01 commented:
Test cases:
1)abc def. ysht u   ->  /abc-def.ysht-u
2)ball#$%.@#$%    ->    /ball.
3)test &^$name.^%$#%test name(&^  ->    /test-name.test-name
4)na* me*$#..jsbf&^...       ->      /na-me.jsbf.
5)test name&%#test name.test name bshdye.hdsbd jasjdb.asjdsabd.anjcxbj*^$#$%^&* ansdbu    
->
   /test-nametest-name.test-name-bshdye.hdsbd-jasjdb.asjdsabd.anjcxbj-ansdbu

MarekB01 commented:
the last commit leaves all the dots which were entered

MarekB01 committed:
BroadleafCommerce/QA-5001_generated_url_issue

Rewrote convertToUrlFragment, so now if product name contains special symbols and "." between it, it'll be removed in generated url field. Updated code to not remove the DOT from the product name. Refactored code, updated it due to requirements. This commit leaves all the dots

MarekB01 committed:
BroadleafCommerce/QA-5001_generated_url_issue

Rewrote convertToUrlFragment, so now if product name contains special symbols and "." between it, it'll be removed in generated url field. Updated code to not remove the DOT from the product name. Refactored code, updated it due to requirements. This commit leaves all the dots. Changed the dots on an underscore.

MarekB01 commented:
@agustinlacombadev Hi, Can you please review the last changes it were developed due to Daniel's comments.
Please test it and let me know if it works fine.
Thanks

agustinlacombadev reviewed (approved):
LGTM - we have been testing it and works as expected

MarekB01 commented:
@riteshadhikari17 I think this one can be merged as it is delivered to customer
